packages/labs/task/src/task.ts

```diff
@@ -191,15 +213,20 @@ export class Task<
    */
   async run(args?: T) {
     args ??= this._getArgs?.();
+
+    // Clear the last complete task in initial because it may be a resolved
+    // promise. Also clear if complete or error because the value returned by
+    // awaiting taskComplete may have changed since last run.
     if (
       this.status === TaskStatus.COMPLETE ||
-      this.status === TaskStatus.ERROR
+      this.status === TaskStatus.ERROR ||
+      this.status === TaskStatus.INITIAL
```

Comment:
This is the same as `this.status !== TaskStatus.PENDING` now, yeah?

Which brings up an interesting question: is `run()` doing the correct thing when called on an already pending task? No need to address that in the PR, but maybe we're missing a test?